src/components/Publish/_utils.ts

```diff
@@ -479,11 +468,16 @@ export async function getFeesTokensAndPricing(
   }
 
   const gasPrice = await web3.eth.getGasPrice()
+  const userBalance = await web3.eth.getBalance(accountId)
 
   const gasFeeEth = web3.utils.fromWei(
     (+gasPrice * +gasEstimate).toString(),
     'ether'
   )
+  const balanceEth = web3.utils.fromWei(userBalance.toString(), 'ether')
```

Comment:
same here with different native tokens on different chains, most likely we have to adapt this here too

Reply:
one idea apart from contract calls would be to use our `useNetworkMetadata()` as have info about all chains within our codebase. The native token for each chain should be in there somewhere

Reply:
can this be achieve with `web3.utils.fromWei(await web3.eth.getBalance(accountId, 'latest'))`, right? like the `getUserBalance` helper in `Web3.tsx`

Reply:
> 

got it. I'll try `useNetworkMetadata`. 

I don't know what's going on with the comments, but i'm not getting them in real time 😂 I'm seen your answers always late

Reply:
yup, just checked, `await web3.eth.getBalance(accountId, 'latest')` works and will return native token balance for each chain. So would be enough unless we require the native token symbol somewhere, then the `useNetworkMetadata()` would be handy

Reply:
also, just seeing that our Web3 context provider holds and refetches user balance and under `balance.eth` will always put the native token balance. So within components we can always use `const { balance } = useWeb3()` to limit excessive balance checks